pkg/sql/sqlbase/sequence.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sqlbase
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
+)
+
+// sequenceFuncs is a set of sequence functions that take in a sequence name
+// (string) as an argument and can be used in a scalar expression.
+// TODO(richardjcai): When implicit casting is supported, these builtins
+// should take RegClass as the arg type for the sequence name instead of
+// string, we will add a dependency on all RegClass types used in a view.
+var sequenceFuncs = map[string]struct{}{
+	"nextval": {},
+	"currval": {},
+}
+
+// GetSequenceFromFunc extracts a sequence name from a FuncExpr if the function
+// takes a sequence name as an arg. Returns the name of the sequence and a bool
+// representing whether a sequence name was found or not.
+func GetSequenceFromFunc(funcExpr *tree.FuncExpr) (string, bool, error) {
+	searchPath := sessiondata.SearchPath{}
```

Comment:
should we be using the search path of the caller here?

Reply:
> So.. if `nextval('s')` is inside a CASE and it wouldn't necessarily get executed, do we want to error out if the source doesn't exist? E.g. this works in postgres:
> 
> ```
> radu=> CREATE VIEW v AS SELECT CASE WHEN EXISTS (SELECT * FROM t WHERE x=1) THEN nextval('s') ELSE 0 END;
> CREATE VIEW
> radu=> SELECT * FROM v;
> ERROR:  "s" is not a sequence
> radu=> DELETE FROM t WHERE true;
> DELETE 1
> radu=> SELECT * FROM v;
>  case 
> ------
>     0
> (1 row)
> ```
> 
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50103#-:-M9ZBoe71Ftt2vRdQA-n:bki1302)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @RichardJCai)

So would it be correct if we didn't error out if the source didn't exist?

The logic I have here should match PG's behaviour right now.

```
root@127.0.0.1:56309/movr> create table t(x int);
CREATE TABLE

Time: 2.623ms

root@127.0.0.1:56309/movr> CREATE VIEW v AS SELECT CASE WHEN EXISTS (SELECT x FROM t WHERE x=1) THEN nextval('s') ELSE 0 END;
CREATE VIEW

Time: 24.966ms

root@127.0.0.1:56309/movr> insert into t values (1);
INSERT 1

Time: 1.017ms

root@127.0.0.1:56309/movr> select * from v;
ERROR: nextval(): relation "s" does not exist
SQLSTATE: 42P01
root@127.0.0.1:56309/movr>
```

Reply:
For resolving in functions, the searchPath isn't actually used.